packages/ra-core/src/controller/button/useDeleteWithConfirmController.tsx

```diff
@@ -72,25 +73,36 @@ const useDeleteWithConfirmController = <RecordType extends RaRecord = any>(
         mutationMode,
         onClick,
         mutationOptions = {},
-        successMessage = 'ra.notification.deleted',
+        successMessage,
     } = props;
     const { meta: mutationMeta, ...otherMutationOptions } = mutationOptions;
     const resource = useResourceContext(props);
     const [open, setOpen] = useState(false);
     const notify = useNotify();
     const unselect = useUnselect(resource);
     const redirect = useRedirect();
+    const translate = useTranslate();
+
     const [deleteOne, { isPending }] = useDelete<RecordType>(
         resource,
         undefined,
         {
             onSuccess: () => {
                 setOpen(false);
-                notify(successMessage, {
-                    type: 'info',
-                    messageArgs: { smart_count: 1 },
-                    undoable: mutationMode === 'undoable',
-                });
+                notify(
+                    successMessage ??
+                        `resources.${resource}.notifications.deleted`,
```

Comment:
Why do you need to set a default here and not in https://github.com/marmelab/react-admin/pull/10203/files#diff-4bd2e2a2759d2f305fc5f5ea2b21a951f1b1721eee2f4519eac68502625771dfR91?